src/jdk.crypto.ec/share/classes/sun/security/ec/SunEC.java

```diff
@@ -60,10 +60,23 @@ public final class SunEC extends Provider {
 
     private static final long serialVersionUID = -2279741672933606418L;
 
+    // flag indicating whether the full EC implementation is present
+    // (when native library is absent then fewer EC algorithms are available)
+    private static boolean useFullImplementation = true;
+
+    private static final boolean nativeCryptTrace = NativeCrypto.isTraceEnabled();
+
+
+    /* The property 'jdk.nativeECKeyGen' is used to control enablement of the native
+     * ECKeyGeneration implementation.
+     * OpenSSL 1.1.0 or above is required for EC key generation support.
+     */
+    private static final boolean useNativeECKeyGen = NativeCrypto.isAlgorithmEnabled("jdk.nativeECKeyGen", "SunEC");
+
     /* The property 'jdk.nativeEC' is used to control enablement of the native
-     * EC implementation.
+     * ECDH implementation.
      */
-    private static final boolean useNativeEC = NativeCrypto.isAlgorithmEnabled("jdk.nativeEC", "SunEC");
+    private static final boolean useNativeECDH = NativeCrypto.isAlgorithmEnabled("jdk.nativeEC", "SunEC");
```

Comment:
I think `jdk.nativeEC=false` should disable all native EC paths, so this needs to be reordered (keeping comments):
```java
    private static final boolean useNativeEC = NativeCrypto.isAlgorithmEnabled("jdk.nativeEC", "SunEC");
    private static final boolean useNativeECKeyGen = useNativeEC && NativeCrypto.isAlgorithmEnabled("jdk.nativeECKeyGen", "SunEC");
```
Notice the continued use of `useNativeEC` instead of `useNativeECDH`, which is more appropriate given the property names.

Reply:
I was talking to @jasonkatonica about this and he was telling me that it would be better if we had more granularity, regarding the ability to disable features. 

I could have the `useNativeEC` disable everything and then have two separate properties for key generation and key agreement. Would that make more sense?

Reply:
It is a good idea to have separate flags to turn on and off individual algorithms for native code paths instead of one blanket flag to turn off all EC related code paths. This is because `EC` and `ECDH` are different cases in field issue scenarios that we would want to turn off in customer environments. For example, we might want to turn off EC key generation but keep ECDH key exchange enabled.

The code originally wanted to rename the blanket EC algorithm flag ( `jdk.nativeEC` ) to something more specific/ appropriate, but we can certainly live with the `jdk.nativeECKeyGen` referring to EC key generation and then `jdk.nativeEC` referring to `ECDH` key exchange.

It is not worth having one flag to disable all EC algorithms (both ECDH and EC key generation) and also individual flags to turn off features on top of that. This seems a bit harder to manage.

For any users who are currently running with `jdk.nativeEC` set to `false`, we would not necessarily want to turn off this new feature in those environments as they would not benefit from the tested performance improvements for EC key generation.

Reply:
Did as agreed and put `jdk.nativeEC` property first.